lib/src/ruby/ruby.cc

```diff
@@ -18,10 +46,21 @@ namespace facter { namespace ruby {
         return true;
     }
 
-    void load_custom_facts(collection& facts, vector<string> const& paths)
+    void load_custom_facts(collection& facts, bool initialize_puppet, vector<string> const& paths)
     {
         module mod(facts, paths);
+        if (initialize_puppet) {
+            try {
+                eval(load_puppet);
```

Comment:
Have you checked that this works? You're setting `$LOAD_PATH` after `initialize_search_paths` is called from the `module` constructor. You may need to do a `ruby_reset` as well to pick up custom facts in `libdir`.